### ISSUE TITLE
Bug: paths misreported when running features concurrently

### ISSUE THREAD

**Issue by kleijnweb** [cucumber/godog#181]
```
$ godog --stop-on-failure -c 10 --tags @wip10 -f progress
.......................................F---- 44


--- Failed steps:

  Scenario: <redacted> # features/manage/topical.feature:32
    Then <redacted> # features/manage/topical.feature:15
      Error: got 404 status for <redacted>


8 scenarios (7 passed, 1 failed)
44 steps (39 passed, 1 failed, 4 skipped)
1m8.331938003s
```

The only thing correct here is line 32 (in the file the scenario is actually in) but then in a completely different file, one that wasn't even matched by the tag filter. Line 15 doesn't make any sense at all, in the file mentioned (which again, does not match the filter) L15 is a table line, and in the actual scenario it is the `Scenario: ` line 2 scenarios before the scenario failing here.

**Comment by l3pp4rd** [comment]
yes, mentioned in another issue. it is due to gherkin, nodes do not have relationship to parent node, so the state needs to be stored. there needs to be another layer on top of gherkin or the gherkin library adapted to maintain the relation.

**Comment by kumarharsh** [comment]
Yup, just enabled concurrency and bumped into this weird issue. I spent hours tracking where I had written those non-existent tests 😅

**Note**: For anyone facing the same issue, you can make this go away by setting `concurrency=1` until this bug is fixed.

**Comment by kumarharsh** [comment]
For refernce, this issue has some more information: https://github.com/DATA-DOG/godog/issues/161. Maybe close this one?

**Comment by kargakis** [comment]
> **Note**: For anyone facing the same issue, you can make this go away by setting `concurrency=1` until this bug is fixed.

Just bumped into this issue, it sucks to have to disable running tests in parallel in order to work around this issue.

**Comment by lonnblad** [comment]
I believe this is related to #161, which I believe has been solved, I ran the code in #161 and it behaves correctly with the latest version of godog.
